src/types/artifact.ts

```diff
@@ -19,3 +19,7 @@ export function isMarkdownArtifactData(data: ArtifactData): data is MarkdownArti
 export function isTableArtifactData(data: ArtifactData): data is TableArtifactData {
   return Array.isArray(data)
 }
+
+export function isSerializedArtifactData(type: string, data: ArtifactData): data is string {
+  return type !== 'markdown' && type !== 'result' && typeof data === 'string'
```

Comment:
I'm not totally sure what values are in ArtifactData but is there a positive check we could make here? Rather than checking if its not markdown or result can we check that it IS another type? I feel like this is fragile if we add more types where the data is a string but its not serialized.

Reply:
You know I think I'm going to drop this type guard and just check if the data is already a string. If it's not I'll serialize it myself so we can use it in code highlighters